### PR TITLE
Fix 19 cmake warnings

### PR DESCRIPTION
**Pull request by chutsu**
This PR is in relation to #19 

In summary this PR just adds a `-Wall` to `CMAKE_CXX_FLAGS` during compilation

### CODE REVIEW

File: wave_utils/src/config.cpp [hunk 1]
```diff
@@ -66,7 +66,7 @@ int ConfigParser::checkVector(std::string key,
     // clang-format on
 
     // check number of values
-    if (this->root[key].size() != vector_size) {
+    if (this->root[key].size() != (size_t) vector_size) {
```
Comment (leokoppel) [inline wave_utils/src/config.cpp]: Use C++ cast. Or change the type of `vector_size`.

File: wave_utils/src/config.cpp [hunk 2]
```diff
@@ -122,6 +122,7 @@ int ConfigParser::loadPrimitive(ConfigParam param) {
         case FLOAT: *param.f = node.as<float>(); break;
         case DOUBLE: *param.d = node.as<double>(); break;
         case STRING: *param.s = node.as<std::string>(); break;
+        default: return -3;
```
Comment (leokoppel) [inline wave_utils/src/config.cpp]: What is -3? Use enum if it has some meaning.

File: wave_utils/src/config.cpp [hunk 3]
```diff
@@ -142,30 +143,32 @@ int ConfigParser::loadArray(ConfigParam param) {
     this->getYamlNode(param.key, node);
     switch (param.type) {
         case BOOL_ARRAY:
-            for (int i = 0; i < node.size(); i++) {
+            for (size_t i = 0; i < node.size(); i++) {
```
Comment (leokoppel) [inline wave_utils/src/config.cpp]: Can use range-based for.

File: wave_utils/src/config.cpp (hunk 2 above)
Comment (chutsu) [inline wave_utils/src/config.cpp]: I'll do another issue to write a doc for this soon. In the mean time let's just keep it this way for the PR? (else this goes beyond the scope of this PR)

File: wave_utils/src/config.cpp (hunk 1 above)
Comment (arunxdas) [inline wave_utils/src/config.cpp]: :+1: static_cast<>

File: wave_utils/src/config.cpp (hunk 3 above)
Comment (chutsu) [inline wave_utils/src/config.cpp]: Done

File: wave_utils/src/config.cpp (hunk 1 above)
Comment (chutsu) [inline wave_utils/src/config.cpp]: Done

**Pull request merged**